### ISSUE TITLE
Invoke-Expression ignores the common -ErrorAction parameter

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#19734]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

`Invoke-Expression` unexpectedly ignores the common `-ErrorAction` parameter when applied to _itself_ (as opposed to the string being evaluated happening to contain a command with `-ErrorAction`).

```powershell
# !! Error still prints.
Invoke-Expression 'Get-Item NoSuchFile' -ErrorAction Ignore
```

The suboptimal workaround is to use `2>$null`, though that is the equivalent of `-ErrorAction SilentlyContinue`, not `Ignore`.

Also note:

* `-ErrorVariable` _does_ work.
* Possibly related bug:
  * #10476

### Expected behavior

No output (the non-terminating error should be ignored).

### Actual behavior

The non-terminating error prints.

### Environment data

```powershell
PowerShell 7.4.0-preview.3
```

**Comment by 237dmitry** [comment]
I think errors rise after executing expression:

```
iex "gi noexist"
iex "gi noexist -ea 0"
```

**Comment by mklement0** [comment]
Yes, `-ErrorAction` _inside the string to be evaluated_ works, but that's not the point of this issue - I've updated the initial post to make that clear.

**Comment by TobiasPSP** [comment]
The cmdlet WG reviewed this. Invoke-Expression does not handle exceptions raised by the code it executes, and you would have to handle exceptions yourself inside your custom code. While this behavior may not be ideal and even a bug when it comes to some of its behavior, like the -ErrorVariable support, it has been the original cmdlet behavior ever since this cmdlet surfaced. Adjusting this would cause severe breaking changes. As a remedy, the WG recommends avoiding Invoke-Expression altogether due to its intrinsic injection-type attack vulnerability,  and instead move to safe approaches, i.e. Invoke-Command and scriptblocks.
